main/src/ca/uwaterloo/flix/tools/pkg/ManifestParser.scala

```diff
@@ -295,6 +303,74 @@ object ManifestParser {
     }
   }
 
+  /**
+    * Converts `depUrl` to a String and retrieves the URL for a jar dependency.
+    * Returns an error if it is not formatted correctly.
+    */
+  private def getUrl(depUrl: AnyRef, p: Path): Result[URL, ManifestError] = {
```

Comment:
There is a new http client in Java that is recommended to use:

https://www.baeldung.com/java-9-http-client

Reply:
I'm a little confused about what you want me to change?
Should I not use java.net.URL? Or is it the way I download things in JarPackageManager?